src/classes/TDTM_TriggerHandler.cls

```diff
@@ -182,7 +182,19 @@ public class TDTM_TriggerHandler {
                                 describeObj.getName() + ': ' + thisAction + ' with ' + (newList != null ? newList : oldList).size() + ' records');
                         return classToRun.run(newlist, oldlist, thisAction, describeObj);
                     }
-	        	} else {
+	        	} else if(classInstance instanceof TDTM_RunnableMutable) {
```

Comment:
@rogeramitchell I'm confused on this. It looks like this `if` is in the `else` that would run the old code. I would expect the check of `} else if (classInstance instanceof TDTM_RunnableMutable) {` to be after the call to `runAsync()`.

Reply:
@force2b it's helpful to see the full extent of the code here. Existing classes that implement `TDTM_Runnable` will fall into the `if` block on line 167, and will either have `runAsync` called or the instance of that class's `run` method. The `else if` block handles classes implementing the new `TDTM_RunnableMutable` interface, which only has one method `run`.

```apex
    private static TDTM_Runnable.DmlWrapper runClass(SObject classToRunRecord, List<Sobject> newList, List<Sobject> oldList, 
    TDTM_Runnable.Action thisAction, Schema.DescribeSobjectResult describeObj, TDTM_Runnable.DmlWrapper dmlWrapper) {
    	        
        if(classToRunRecord != null) {
	        String classToRunName = String.valueOf(classToRunRecord.get('Class__c'));
	        Type classType = Type.forName(classToRunName);
	        
	        if(classType != null) {   	
	        	Object classInstance = classType.newInstance();
	        	
	        	//The objects we create dynamically need to implement the TDTM_Runnable interface
	        	if(classInstance instanceof TDTM_Runnable) {
			        TDTM_Runnable classToRun = (TDTM_Runnable)classInstance;
			        Boolean async = Boolean.valueOf(classToRunRecord.get('Asynchronous__c'));
			        boolean isBefore = (thisAction == TDTM_Runnable.Action.BeforeInsert || 
			             thisAction == TDTM_Runnable.Action.BeforeUpdate || thisAction == TDTM_Runnable.Action.BeforeDelete); 
			        
			        //A class can only run asynchronously (calling a future) if we have the ids of the
                    //records we want to affect. This means we can only run it for after in insert. 
                    //Also,a future cannot be called from a future or batch method.
                    if(async && !isBefore && !System.isFuture() && !System.isBatch()) {
                        UTIL_Debug.debug(LoggingLevel.WARN, '****Calling asynchronously: ' + classToRunName + ' for ' +
                                describeObj.getName() + ': ' + thisAction + ' with ' + (newList != null ? newList : oldList).size() + ' records');
                        runAsync(classToRun, classToRunName, newList, oldList, thisAction, describeObj);
                    } else {
                        UTIL_Debug.debugWithInfo('****Calling synchronously: ' + classToRunName + ' for ' +
                                describeObj.getName() + ': ' + thisAction + ' with ' + (newList != null ? newList : oldList).size() + ' records');
                        return classToRun.run(newlist, oldlist, thisAction, describeObj);
                    }
	        	} else if(classInstance instanceof TDTM_RunnableMutable) {
                    // alternatively, objects created dynamically can implement the TDTM_RunnableMutable
                    // interface if the class intends to directly modify the global DmlWrappper
                    TDTM_RunnableMutable classToRun = (TDTM_RunnableMutable) classInstance;
                    UTIL_Debug.debugWithInfo('****Calling synchronously: ' + classToRunName + ' for ' +
                        describeObj.getName() + ': ' + thisAction + ' with ' + (newList != null ? newList : oldList).size() + ' records');

                    // we do not check whether class is identified as Asynchronous__c because the
                    // underlying assumption to mutating DmlWrapper is that the class modifies
                    // the collection directly, and TDTM_TriggerHandler.run will perform DML on
                    // whatever exists in the collection after iterating through all classes to run
                    classToRun.run(newList, oldList, thisAction, describeObj, dmlWrapper);
                } else {
	        		UTIL_Debug.debug(LoggingLevel.WARN, '****The class does not implement the required interface.');
	        	}
	        } else {
	        	UTIL_Debug.debug(LoggingLevel.WARN, '****ClassType was null.');
	        }
        } else {
        	UTIL_Debug.debug(LoggingLevel.WARN, '****ClassToCall was null');
        }
        return null;
    }
```